extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/HttpConfiguration.java

```diff
@@ -85,6 +85,16 @@
      */
     public BodyConfig body;
 
+    /**
+     * The encryption key that is used to store persistent logins (e.g. for form auth). Logins are stored in a persistent
+     * cookie that is encrypted with AES-256 using a key derived from a SHA-256 hash of the key that is provided here.
+     *
+     * If no key is provided then an in-memory one will be generated, this will change on every restart though so it
+     * is not suitable for production environments. This must be more than 16 characters long for security reasons
+     */
+    @ConfigItem(name = "auth.session.encryption-key")
+    public String encryptionKey;
```

Comment:
@stuartwdouglas should an `auth.session.encryption-key-location` be added and recommended by default ? `encryptionKey` is a good start but I'm not sure it is good for the production either, unless we have #3937, which would be good, as the users would just type 
```
auth.session.encryption-key=${my.encryption.key.from.vault}
```

Reply:
You can just set the QUARKUS_HTTP_AUTH_SESSION_ENCRYPTION_KEY env var and it will override.

Reply:
Either way this can be done as a future enhancement if there is demand for it. FORM is supposed to be mostly for learning/small apps, for big apps keycloak is our recommended solution.

Reply:
@stuartwdouglas That is fine. I'll create an issue to follow up. Was not really my intention to slow down the PR - but if we let the users to for ex provide the user passwords with the `elytron-security-jdbc` then there should be a way for them not put the secret key for protecting the cookie data in the clear form. But as I said, with Vault and the config indirection it will already be possible; adding a simple check to load it from a (hopefully secure) file system will just complement it.